src/tests/banners/Channel2BannerTests.ts

```diff
@@ -0,0 +1,60 @@
+import fetch from 'node-fetch';
+import {
+    BannerTemplate,
+    BannerTestGenerator,
+    BannerTest,
+    RawTestParams,
+    RawVariantParams,
+    BannerVariant,
+} from '../../types/BannerTypes';
+import { ContributionsBannerPath } from './ContributionsBannerTests';
+import { isProd } from '../../lib/env';
+
+export const DigitalSubscriptionsBannerPath = 'digital-subscriptions-banner.js';
+export const GuardianWeeklyBannerPath = 'guardian-weekly-banner.js';
+
+export const BannerPaths: { [key in BannerTemplate]: string } = {
+    [BannerTemplate.ContributionsBanner]: ContributionsBannerPath,
+    [BannerTemplate.DigitalSubscriptionsBanner]: DigitalSubscriptionsBannerPath,
+    [BannerTemplate.GuardianWeeklyBanner]: GuardianWeeklyBannerPath,
+};
+
+const Channel2BannerContentUrl = isProd
+    ? 'https://gu-contributions-public.s3-eu-west-1.amazonaws.com/banner/PROD/banner-tests2.json'
+    : 'https://gu-contributions-public.s3-eu-west-1.amazonaws.com/banner/CODE/banner-tests2.json';
+
+const Channel2BannerVariant = (variant: RawVariantParams): BannerVariant => {
+    return {
+        name: variant.name,
+        modulePath: BannerPaths[variant.template],
+        moduleName: variant.template,
+        bannerContent: {
+            messageText: variant.body,
+            heading: variant.heading,
+            cta: variant.cta,
+            secondaryCta: variant.secondaryCta,
+        },
+    };
+};
+
+const Channel2BannerTest = (testParams: RawTestParams): BannerTest => {
+    return {
+        name: testParams.name,
+        bannerType: 'subscriptions',
+        testAudience: testParams.userCohort,
+        locations: testParams.locations,
+        canRun: (): boolean => testParams.isOn,
+        minPageViews: testParams.minArticlesBeforeShowingBanner,
+        variants: testParams.variants.map(Channel2BannerVariant),
+        componentType: 'ACQUISITIONS_SUBSCRIPTIONS_BANNER',
```

Comment:
The `componentType` now needs to vary depending on the `template`.
Let's add `componentType: OphanComponentType` to the `BannerVariant` model, then use `variant.componentType` here for the tracking:
https://github.com/guardian/support-dotcom-components/blob/6705e267571fa44917b56fc074e206a00768833c/src/server.tsx#L173

Reply:
Makes sense, it looks like we also pass an array of OphanProducts with the subscriptions banners so I'll add that in too
https://github.com/guardian/support-dotcom-components/blob/6705e267571fa44917b56fc074e206a00768833c/src/tests/banners/DigitalSubscriptionsBannerTest.ts#L28